eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/api/EclairApi.scala

```diff
@@ -98,6 +103,8 @@ trait EclairApi {
     getNodeURI.map(_.nodeId)
   }
 
+  def createInvoice(description: String, amountMsat: Option[MilliSatoshis], expireIn: Option[Long], fallbackAddress: Option[Address], paymentPreimage: Option[String]): Future[LnInvoice]
```

Comment:
I think `expireIn` should be of type `Option[FiniteDuration]`, and I think we should create a new type called `PaymentPreimage` in the `core/` project to represent a pre-image. It probably should be located here: 

https://github.com/bitcoin-s/bitcoin-s/tree/master/core/src/main/scala/org/bitcoins/core/protocol/ln

and I think it should be

```scala

/** Scaladoc blah blah blah... */ 
case class PaymentPreimage(bytes: ByteVector) extends NetworkElement { 
//can we put any invariants here, like does `bytes` need to be 32 bytes in size? 

lazy val hash: Sha256Digest = CryptoUtil.sha256Digest(bytes)
}

object PaymentPreimage extends Factory[PaymentPreimage] { 
  override def fromBytes(bytes: ByteVector): PaymentPreimage = { 
    new PaymentPreimage(bytes)  
  }

}